src/libsystemd/sd-journal/journal-file.h

```diff
@@ -189,7 +189,7 @@ static inline bool VALID_EPOCH(uint64_t u) {
 
 static inline uint64_t journal_file_entry_seqnum(JournalFile *f, Object *o) {
         return JOURNAL_HEADER_COMPACT(f->header)
-                        ? le64toh(o->entry.compact.seqnum)
+                        ? le32toh(o->entry.compact.seqnum) + le64toh(f->header->head_entry_seqnum)
```

Comment:
hmm, overflow check in case rogue file as super high seqnum set in header? not sure what a good strategy is in this case though, maybe return an invalid seqnum (i.e. UINT64_MAX) and then catch that everywhere and make it an EBADMSG issue